lib/internal/modules/esm/translators.js

```diff
@@ -103,31 +103,34 @@ translators.set('builtin', async (url) => {
     });
 });
 
-// Strategy for loading a node native module
-translators.set('addon', async (url) => {
-  debug(`Translating NativeModule ${url}`);
-  return createDynamicModule(['default'], url, (reflect) => {
-    debug(`Loading NativeModule ${url}`);
-    const module = { exports: {} };
-    const pathname = internalURLModule.fileURLToPath(new URL(url));
-    process.dlopen(module, _makeLong(pathname));
-    reflect.exports.default.set(module.exports);
-  });
-});
-
 // Strategy for loading a JSON file
-translators.set('json', async (url) => {
+translators.set('json', async function jsonStrategy(url) {
   debug(`Translating JSONModule ${url}`);
-  return createDynamicModule(['default'], url, (reflect) => {
-    debug(`Loading JSONModule ${url}`);
-    const pathname = internalURLModule.fileURLToPath(new URL(url));
-    const content = readFileSync(pathname, 'utf8');
-    try {
-      const exports = JsonParse(stripBOM(content));
+  debug(`Loading JSONModule ${url}`);
+  const pathname = fileURLToPath(url);
+  const modulePath = isWindows ?
+    StringReplace(pathname, winSepRegEx, '\\') : pathname;
+  let module = CJSModule._cache[modulePath];
+  if (module && module.loaded) {
+    const exports = module.exports;
+    return createDynamicModule(['default'], url, (reflect) => {
       reflect.exports.default.set(exports);
-    } catch (err) {
-      err.message = pathname + ': ' + err.message;
-      throw err;
-    }
+    });
+  }
+  const content = await readFileAsync(pathname, 'utf-8');
+  try {
+    const exports = JsonParse(stripBOM(content));
+    module = {
+      exports,
+      loaded: true
+    };
+  } catch (err) {
+    err.message = pathname + ': ' + err.message;
+    throw err;
```

Comment:
As a suggestion that could also be implemented later on:

We could add a NodeCore error that wraps the JSON parse error instead of just manipulating the original error message. That would allow to add further properties and maybe additional debugging information.

Reply:
added a TODO